Frameworks/CoreText/DWriteWrapper.mm

```diff
@@ -0,0 +1,392 @@
+//******************************************************************************
+//
+// Copyright (c) 2016 Microsoft Corporation. All rights reserved.
+//
+// This code is licensed under the MIT License (MIT).
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+//******************************************************************************
+
+#include <COMIncludes.h>
+#import <wrl/implements.h>
+#include <COMIncludes_End.h>
+#import <mutex>
+#import <vector>
+#import <algorithm>
+#import <CoreFoundation/CFBase.h>
+#import "CoreTextInternal.h"
+#import "DWriteWrapper.h"
+
+using namespace std;
+using namespace Microsoft::WRL;
+
+static const wchar_t* TAG = L"_DWriteWrapper";
+static const wchar_t* c_defaultUserLanguage = L"en-us";
+static const wchar_t* c_defaultFontName = L"Gabriola";
+
+// TODO::
+// These should be removed once CTFont and UIFont are bridged.
+static IWLazyClassLookup s_lazyUIFont("UIFont");
+static const float c_defaultSystemFontSize = 15.0f;
+
+static ComPtr<IDWriteFactory> CreateDWriteFactoryInstance() {
+    ComPtr<IDWriteFactory> dwriteFactory;
+    THROW_IF_FAILED(DWriteCreateFactory(DWRITE_FACTORY_TYPE_SHARED, __uuidof(IDWriteFactory), &dwriteFactory));
+    return dwriteFactory;
+}
+
+static ComPtr<IDWriteFactory> GetDWriteFactoryInstance() {
```

Comment:
>IDWriteFactory [](start = 14, length = 14)

Nit: If this method is called a lot, it would make sense to simply return the IDWriteFactory* instead.  The caller will save it into a ComPtr, holding reference correctly.  As is, we will have extra addref/release due to the temporary object.